### ISSUE TITLE
Upstream devfiles have missing debug configurations during test runs

### ISSUE THREAD

**Issue by ScrewTSW** [eclipse-che/che#20046]
Related https://github.com/eclipse/che/issues/19880 #19887 

### Describe the bug

There are missing debug configurations (not being loaded at all) in workspaces, until you refresh the page, then they appear.
Example of this is happy-path test runs, they fail at debugging steps:

https://main-jenkins-csb-crwqe.apps.ocp4.prod.psi.redhat.com/job/Che/job/e2e/job/minikube/job/pr-check/job/che-tests-pr-check/85/consoleFull
https://main-jenkins-csb-crwqe.apps.ocp4.prod.psi.redhat.com/job/Che/job/e2e/job/minikube/job/pr-check/job/che-tests-pr-check/85/artifact/che/tests/e2e/report/

* Before page refresh  
![Screenshot_20210630_122656](https://user-images.githubusercontent.com/16451875/123945880-e41b8100-d99e-11eb-9c1b-9f49976abc0a.png)

* After page refresh  
![Screenshot_20210630_122756](https://user-images.githubusercontent.com/16451875/123945924-f09fd980-d99e-11eb-8802-03e79ee2648e.png)

### Che version

 - [x] latest
 - [x] nightly
 - [x] other: 7.32.x

### Steps to reproduce
* Start any workspace, or you can use happy path custom devfile: https://github.com/eclipse/che/blob/main/tests/e2e/files/happy-path/happy-path-workspace.yaml
* Wait for the LS to initialize
* Run all necessary install and build steps
* Launch the application in debug mode
* Open debug panel on the left - no configurations
* Refresh page
* Open debug panel again - configurations present

### Expected behavior
* Debug configurations appear in debug panel once project pull is finished

### Runtime
   - [ ] kubernetes (include output of `kubectl version`)
   - [ ] Openshift (include output of `oc version`)
   - [ ] minikube (include output of `minikube version` and `kubectl version`)
   - [ ] minishift (include output of `minishift version` and `oc version`)
   - [ ] docker-desktop + K8S (include output of `docker version` and `kubectl version`)
   - [x] other: CRW QE OCP 4.7 AWS

### Screenshots
in description

### Installation method
   - [x] chectl
     * provide a full command that was used to deploy Eclipse Che:
     ```
     chectl server:deploy -n tdancs-che-nightly -p openshift --no-olm-suggested-namespace --che-operator-cr-patch-yaml=/home/tdancs/Workspace/oauthless_patch.yaml
     ```
  
     ```
     $ cat /home/tdancs/Workspace/oauthless_patch.yaml 
     spec:
       auth:
         openShiftoAuth: false
         updateAdminPassword: false
     ```
     * provide an output of `chectl version` command:
     ```
     chectl/0.0.20210629-next.ff17d27 linux-x64 node-v12.22.1
     ```
   - [ ] OperatorHub
   - [ ] I don't know

### Environment
   - [ ] my computer
       - [ ] Windows
       - [ ] Linux
       - [ ] macOS
   - [x] Cloud
       - [ ] Amazon
       - [ ] Azure
       - [ ] GCE
       - [x] other - CRW QE OCP 4.7 AWS cluster
   - [ ] Dev Sandbox (workspaces.openshift.com)
   - [ ] other: please specify

### Eclipse Che Logs

N/A

### Additional context

N/A

**Comment by dmytro-ndp** [comment]
The issue could probably duplicate https://github.com/eclipse/che/issues/18034

**Comment by ScrewTSW** [comment]
@dmytro-ndp it is related, but this is not a test failure, this is a product/devfile failure

**Comment by dmytro-ndp** [comment]
@ScrewTSW : thank you,
I am going to add workaround to Happy path tests - to refresh the page in case of problem occurred.

**Comment by dmytro-ndp** [comment]
@ScrewTSW : there is also another issue similar to this one: https://github.com/eclipse/che/issues/19887
